lib/index.js

```diff
@@ -572,12 +587,29 @@ internals.getJoinTableInfo = (model, knex, cb) => {
                         skippedCols.push({
                             tableName,
                             column,
-                            type
+                            type,
+                            schemaConflict: false // Set as a default, may be overridden below
                         });
                     }
                     return !err;
                 });
 
+            const joiKeysToStrip = skippedCols.map((sc) => sc.column);
+
+            // Update skippedCols to say there's a schemaConflict
+            const onKeyStripped = (keyName) => {
+
+                skippedCols.find((sc) => sc.column === keyName).schemaConflict = true;
+            };
```

Comment:
Modifying this object here instead of dependency-injecting it, how do we feel about that?

Reply:
This whole thing generally doesn't sit well with me.  I was a little surprised it worked at all, but what happens if `find()` doesn't find anything?

Reply:
Outside of context and having not written it, I can see why that find would look scary. In here only keys from `joiKeysToStrip` will be passed to `onKeyStripped`, and joiKeysToStrip is filled with column names from `skippedCols`